src/java/org/apache/cassandra/index/sai/disk/v1/MergingIterator.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.v1;
+
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import org.apache.cassandra.index.sai.disk.TermsIterator;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.lucene.util.PriorityQueue;
+
+public final class MergingIterator implements Iterator<ByteComparable>
```

Comment:
We couldn't use `org.apache.cassandra.utils.MergeIterator`, because we're not only merging terms, but merging postings for those terms as we go, correct? That makes sense, but I'd almost want to move it into `TermsIteratorMerger`, either in its current state or w/ its state just inlined. One doesn't seem to make sense without the other.

Reply:
Agreed, I'll move it into `TermsIteratorMerger` but keep its current form as a separate class.